src/main/groovy/jenkins/automation/utils/CheckmarxUtils.groovy

```diff
@@ -0,0 +1,172 @@
+ *               Checkmarx job in the Jenkins UI
+ * <br/>&#x2014; <code>useOwnServerCredentials</code>: optional, defaults to
+ *               <code>false</code>, specifies whether to use the global or per-job
+ *               Checkmarx server URL/credentials
+ * <br/>&#x2014; <code>serverUrl</code>: optional, specifies the Checkmarx server
+ *               URL to be used
+ * <br/>&#x2014; <code>username</code>: optional, specifies the Checkmarx username
+ * <br/>&#x2014; <code>password</code>: optional, specifies the Checkmarx password
+ * <br/>&#x2014; <code>excludeFolders</code>: optional, see code for the default,
+ *               specifies the directories to exclude from scanning
+ * <br/>&#x2014; <code>filterPattern</code>: optional, see code for the default,
+ *               specifies globs to filter out
+ * <br/>&#x2014; <code>incremental</code>: optional, defaults to <code>true</code>,
+ *               specifies whether scan should be incremental
+ * <br/>&#x2014; <code>fullScanCycle</code>: optional, defaults to <code>10</code>,
+ *               specifies how frequently to run a full scan when normally doing
+ *               incremental scans
+ * <br/>&#x2014; <code>comment</code>: optional, additional comments to be adding
+ *               to the scan results
+ * <br/>&#x2014; <code>vulnerabilityThreshold</code>: optional, defaults to
+ *               <code>true</code>, fail the build if the number of vulnerabilities
+ *               exceeds the thresholds set
+ * <br/>&#x2014; <code>highThresholdDefault</code>: optional, defaults to
+ *               <code>1</code>, sets the threshold for max number of 'high'
+ *               vulnerabilities
+ * <br/>&#x2014; <code>mediumThresholdDefault</code>: optional, defaults to
+ *               <code>2</code>, sets the threshold for max number of 'medium'
+ *               vulnerabilities
+ * <br/>&#x2014; <code>lowThresholdDefault</code>: optional, defaults to
+ *               <code>3</code>, set the threshold for max number of 'low'
+ *               vulnerabilities
+ * @see <a href="https://github.com/cfpb/jenkins-automation/blob/gh-pages/docs/examples.md#checkmarx-security-job-builder" target="_blank">Checkmarx job Example</a>
+ *
+ */
+
+    static void checkmarxScan(context, Map cxConfig) {
+        assert cxConfig.projectName != null
+        assert cxConfig.groupId != null
+
+        def useOwnServerCredentialsDefault = false
+        def presetDefault = "17" // "Default 2014"
+        def excludeFoldersDefault = "resources, .git"
+        def incrementalDefault = true
+        def fullScanCycleDefault = "10"
+        def commentDefault = ""
+        def vulnerabilityThresholdDefault = true
+        def highThresholdDefault = 1
+        def mediumThresholdDefault = 2
+        def lowThresholdDefault = 3
+        def filterPatternDefault = """
+            !**/_cvs/**/*, !**/.svn/**/*,   !**/.hg/**/*,   !**/.git/**/*,  !**/.bzr/**/*, !**/bin/**/*,
```

Comment:
Because this is so large and hard to deal with have you considered moving it out of this file into a config file?

Reply:
Not sure about a config 'file', but I can see them being moved out of the method, potentially. Is there an existing pattern here in JAC you can point to along these lines?

Reply:
I experimented with a couple of different methods, and I settled on the one I'm seeing in JAC from HMDA and from Delivery that uses a groovy config file then ConfigSlurper to parse it.

HMDA - https://github.cfpb.gov/HMDA-Operations/hmda-devops/blob/master/jenkins/apps/env.groovy
Delivery uses a config file in `jenkins-automation-private`